src/server/status_server.rs

```diff
@@ -561,6 +599,45 @@ mod tests {
         status_server.stop();
     }
 
+    #[test]
+    fn test_log_level_endpoint() {
+        let mut status_server = StatusServer::new(1, dummy_future_scheduler());
+        let _ = status_server.start("127.0.0.1:0".to_string());
+
+        let log_level_uri = Uri::builder()
+            .scheme("http")
+            .authority(status_server.listening_addr().to_string().as_str())
+            .path_and_query("/log/log-level")
+            .build()
+            .unwrap();
+        let mut log_level_request = Request::new(Body::from(
+            serde_json::to_string(&LogLevelRequest {
+                log_level: slog::Level::Debug,
+            })
+            .unwrap(),
+        ));
+        *log_level_request.method_mut() = Method::PUT;
+        *log_level_request.uri_mut() = log_level_uri;
+        log_level_request.headers_mut().insert(
+            hyper::header::CONTENT_TYPE,
+            hyper::header::HeaderValue::from_static("application/json"),
+        );
+
+        let log_level_handle = status_server.thread_pool.spawn_handle(lazy(move || {
```

Comment:
May need to verify the log level is actually changed.

Reply:
Added an assertion for that